android-components/components/feature/privatemode/src/main/java/mozilla/components/feature/privatemode/notification/AbstractPrivateNotificationService.kt

```diff
@@ -114,11 +121,13 @@ abstract class AbstractPrivateNotificationService : Service() {
      * The service should be started only if private tabs are open.
      */
     final override fun onCreate() {
-        val notificationId = getNotificationId()
-        val channelId = getChannelId()
-        val notification = createNotification(channelId)
+        notificationScope.launch {
+            val notificationId = getNotificationId()
+            val channelId = getChannelId()
+            val notification = createNotification(channelId)
 
-        startForeground(notificationId, notification)
+            startForeground(notificationId, notification)
```

Comment:
Does this work in your testing? I'm pretty sure we need to call `startForeground` on the main thread since services run there.

Reply:
Moved it to the main thread.